prometheus-net/Histogram.cs

```diff
@@ -48,55 +47,35 @@ internal Histogram(string name, string help, string[] labelNames, double[] bucke
 
         public class Child : Advanced.Child, IHistogram
         {
-            private double _sum = 0;
-            private ulong _count = 0;
-            private ulong[] _bucketCounts;
-            private readonly object _lock = new object();
-
-            //re-using WireFormat.Histogram to avoid excessive array allocations 
```

Comment:
Regarding these TODOs

1) Reusing the wire format is wonky because any duplicate request to obtain metrics will be operating off the same instance of the wireMetrics that we mod and return. This means that one request could be in the progress of emitting and rendering these metrics while another request is in the process of collecting and modding that same wireMetrics instance. The implementation was modified to produce a new wire metrics instance on collection call in a similar fashion to the official java client.  Was there any solid evidence that the allocation of arrays was causing a meaningful performance issue?

2) HdrHistogram is not a good candidate here because the library explicitly states that is is not thread-safe. It's possible that some of the optimizations they use could be incorporated here.  Perhaps that could be investigated?